domain/src/main/java/org/oppia/android/domain/onboarding/AppStartupStateController.kt

```diff
@@ -116,11 +133,51 @@ class AppStartupStateController @Inject constructor(
   }
 
   private fun computeAppStartupMode(onboardingState: OnboardingState): StartupMode {
-    return when {
-      hasAppExpired() -> StartupMode.APP_IS_DEPRECATED
-      onboardingState.alreadyOnboardedApp -> StartupMode.USER_IS_ONBOARDED
-      else -> StartupMode.USER_NOT_YET_ONBOARDED
+    // Return old logic if app and os feature flag is not enabled
```

Comment:
Could you instead specify what the **old** logic is?

Reply:
I have defined this better. Let me know if it is up-to-standard or if it requires some extra modification.

Reply:
This LGTM in latest iteration.